common/src/main/java/org/cloudfoundry/identity/uaa/oauth/token/UaaTokenServices.java

```diff
@@ -325,8 +325,7 @@ private OAuth2AccessToken createAccessToken(String userId, String username, Stri
     }
 
     private void populateIdToken(OpenIdToken token, Set<String> scopes, Set<String> responseTypes) {
-        if (scopes.contains("openid") &&
-            responseTypes.contains(OpenIdToken.ID_TOKEN)) {
+        if (scopes.contains("openid")) {
```

Comment:
The check here must be for the 'scope' HTTP request parameter. The scope parameter contains openid we return id_token.
The scopes parameter is a list of all user scopes. We're still looking into how to consolidate this